rpmio/expression.c

```diff
@@ -110,6 +110,12 @@ typedef struct _parseState {
 static void exprErr(const struct _parseState *state, const char *msg,
 		    const char *p)
 {
+    const char *s = state->str;
+
+    s = strchr(s,'\n');
+    if (s && (*(s+1) != '\0'))
+	p = NULL;
+
```

Comment:
While "s" is perfectly acceptable variable name in a case like this, a more descriptive one would make the code that little bit more readable. Say, "nl" or even "newline", after which the whole thing reads almost as human language. *Almost* :)